app/controllers/messages_controller.rb

```diff
@@ -9,11 +9,15 @@ def new
   def create
     m1 = message_params
     @email = m1[:receiver_id]
-    m1[:receiver_id] = User.find_by_email(@email).id
-    @message = current_user.sent_messages.new(m1)
-    @message.save!
-    if @message.save!
-      redirect_to dashboard_index_path
+    @receiver = User.find_by_email(@email)
+    if @receiver != nil
```

Comment:
also we should not check `!= nil`

Reply:
For absent user User.find_by_email returns nil so was checking for that. What is more appropriate method, if i user user.exists? then there will be extra query.